### PR TITLE
Fix find_ysoserial_offsets.rb and incorrect serialVersionUID patching

### PR DESCRIPTION
**Pull request by wvu**
```
/find_ysoserial_offsets.rb:39:in `generate_payload': undefined local variable or method `ysoserial_modified' for main:Object (NameError)
	from /find_ysoserial_offsets.rb:140:in `block in <main>'
	from /find_ysoserial_offsets.rb:137:in `each'
	from /find_ysoserial_offsets.rb:137:in `<main>'
```

```
Invalid payload type 'Payload'
```

```
Invalid payload type '-------'
```

I did not update `data/ysoserial_payloads.json` because it seems to break the `CommonsBeanutils1` payload used in [exploit/windows/http/desktopcentral_deserialization](https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/windows/http/desktopcentral_deserialization.rb). We will have to address that in a later PR.

Fixes #11125 and #13071. See also: #11359.

### CODE REVIEW

**Review by smcintyre-r7 — APPROVED** [review]

File: lib/msf/util/java_deserialization.rb [hunk 1]
```diff
@@ -54,20 +54,6 @@ def self.ysoserial_payload(payload_name, command=nil, serial_version_uid: nil)
...
-        unless serial_version_uid.is_a?(String)
-          raise ArgumentError, 'serial_version_uid must be a string'
-        end
-
-        if serial_version_uid.length != 8
-          raise ArgumentError, 'serial_version_uid must be 8 bytes'
-        end
-
-        # XXX: This hardcoded value may change!
-        bytes.sub!("\xe3\xa1\x88\xea\x73\x22\xa4\x48", serial_version_uid)
```
Comment (wvu-r7) [inline lib/msf/util/java_deserialization.rb]: This won't work if the `CommonsBeanutils1` payload in `data/ysoserial_payloads.json` is updated.

File: modules/exploits/windows/http/desktopcentral_deserialization.rb [hunk 2]
```diff
@@ -134,10 +134,12 @@ def execute_command(cmd, _opts = {})
     # I identified mr_me's binary blob as the CommonsBeanutils1 payload :)
     serialized_payload = Msf::Util::JavaDeserialization.ysoserial_payload(
       'CommonsBeanutils1',
-      cmd,
-      serial_version_uid: "\xcf\x8e\x01\x82\xfe\x4e\xf1\x7e"
+      cmd
     )
 
+    # XXX: Patch in expected serialVersionUID
+    serialized_payload[140, 8] = "\xcf\x8e\x01\x82\xfe\x4e\xf1\x7e"
```
Comment (wvu-r7) [inline modules/exploits/windows/http/desktopcentral_deserialization.rb]: This should still work if the offset doesn't change after updating the `CommonsBeanutils1` payload in `data/ysoserial_payloads.json`. However, the updated payload seems to be breaking for me regardless.

**Comment by smcintyre-r7** [comment]
Retested with the `windows/http/desktopcentral_deserialization` module and the `tools/payloads/ysoserial./runme.sh` tool. Both are working, with the module yielding a session and the tool updating the JSON data file. Thanks @wvu-r7 ! I'll land this in a moment.

**Comment by smcintyre-r7** [comment]
# Release Notes
This fixes a bug in the tool workflow which generates the data files for YSoSerial Java payloads.

**Pull request merged**